plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/commands/ActionRegistrar.kt

```diff
@@ -15,7 +23,26 @@ class ActionRegistrar {
     val flow = _messages.asSharedFlow()
 
     fun reportMessageClick(command: EditorContextCommand, project: Project) {
-        _messages.tryEmit(ContextMenuActionMessage(command, project))
+        if (command == EditorContextCommand.GenerateUnitTests) {
+            // pre-existing old chat code path
+            _messages.tryEmit(ContextMenuActionMessage(command, project))
+        } else {
+            // new agentic chat route
+            runBlocking {
```

Comment:
this will block edt

Reply:
well if I mark `reportMessageClick` as suspend it would require me to mark other functions as `suspend`. 

Should I launch with coroutine?

Reply:
i really don't like ActionRegistrar
let's just do
```
ApplicationManager.getApplication().executeOnPooledThread {
    runBlocking {
...
    }
}
```

Reply:
maybe try this https://youtrack.jetbrains.com/issue/IJPL-371/Provide-currentThreadCoroutineScope-inside-AnActionactionPerformed

you can request a coroutine scope from the platform in the action system

Reply:
maybe like
```kt
        currentThreadCoroutineScope().launch(getCoroutineBgContext()) {
            ActionRegistrar.instance.reportMessageClick
        }
```

Reply:
We can skip ActionRegistrar for these and handle it in custom action

Reply:
>skip ActionRegistrar for these and handle it in custom action
This would require more lines of code.